src/config/scrivitoSites.ts

```diff
@@ -43,23 +44,35 @@ export function siteForUrl(
     return { baseUrl: neoletterBaseUrl, siteId: NEOLETTER_MAILINGS_SITE_ID }
   }
 
-  const language = languageForUrl(url)
-  const websites = appWebsites()
-  const siteId = websites
-    ?.find((site) => site.language() === language)
-    ?.siteId()
+  const { contentId, language, siteId } = findSiteByUrl(url)
 
-  if (language && siteId) {
-    return { baseUrl: `${getBaseAppUrl()}/${language}`, siteId }
-  }
-  if (websites?.length) return findSiteForUrlExpensive(url)
+  if (siteId) return { baseUrl: baseUrlFor(language, contentId), siteId }
+
+  return findSiteForUrlExpensive(url)
 }
 
-function languageForUrl(url: string) {
-  const regex = new RegExp(
-    `^${getBaseAppUrl()}\\/(?<lang>[a-z]{2}(-[A-Z]{2})?)([?/]|$)`,
-  )
-  return regex.exec(url)?.groups?.lang
+function findSiteByUrl(url: string) {
+  const { contentId: urlContentId, language } = extractFromUrl(url)
+  if (!language) return {}
+
+  if (urlContentId === defaultSiteContentId()) return {}
+
+  const contentId = urlContentId || defaultSiteContentId()
+  if (!contentId) return {}
+
+  const siteId = findSiteIdBy({ contentId, language })
+  if (!siteId) return {}
+
+  return { contentId, language, siteId }
+}
+
+function findSiteIdBy(query: { contentId: string; language: string }) {
+  return Obj.onAllSites()
+    .where('_path', 'equals', '/')
+    .and('_contentId', 'equals', query.contentId)
+    .and('_language', 'equals', query.language)
+    .toArray()[0]
+    ?.siteId()
```

Comment:
FYI: In a follow up we can try to improve the loading behavior, so that we only execute the minimum amount of search requests related to site finding/picking.